app/models/planning_application.rb

```diff
@@ -482,6 +478,36 @@ def create_audit!
     audit_created!(activity_type: "created", activity_information: Current.api_user&.name || Current.user&.name)
   end
 
+  def audit_updated!
+    if saved_changes?
+      saved_changes.keys.intersection(PLANNING_APPLICATION_PERMITTED_KEYS).map do |attribute_name|
+        next if saved_change_to_attribute(attribute_name).all?(&:blank?)
+
+        attribute_to_audit(attribute_name)
+      end
+    end
+  end
+
+  def attribute_to_audit(attribute_name)
+    if attribute_name.eql?("constraints")
+      audit_constraits!(saved_changes)
```

Comment:
I have a feeling we should be creating a class to help with this. Probably out of scope for now but there's probably enough complexity to warrant this